src/main/python/org/broadinstitute/hellbender/tools/gcnv/bin/case_denoising_calling.py

```diff
@@ -0,0 +1,210 @@
+
+# logging args
+gcnvkernel.cli_commons.add_logging_args_to_argparse(parser)
+
+# add tool-specific args
+group = parser.add_argument_group(title="Required arguments")
+
+group.add_argument("--input_model_path",
+                   type=str,
+                   required=True,
+                   default=argparse.SUPPRESS,
+                   help="Path to denoising model parameters")
+
+group.add_argument("--read_count_tsv_files",
+                   type=str,
+                   required=True,
+                   nargs='+',  # one or more
+                   default=argparse.SUPPRESS,
+                   help="List of read count files in the cohort (in .tsv format; must include sample name header)")
+
+group.add_argument("--ploidy_calls_path",
+                   type=str,
+                   required=True,
+                   default=argparse.SUPPRESS,
+                   help="The path to the results of ploidy determination tool")
+
+group.add_argument("--output_calls_path",
+                   type=str,
+                   required=True,
+                   default=argparse.SUPPRESS,
+                   help="Output path to write CNV calls")
+
+group.add_argument("--output_opt_path",
+                   type=str,
+                   required=False,
+                   default=argparse.SUPPRESS,
+                   help="(advanced) Output path to write the latest optimizer state")
+
+group.add_argument("--input_calls_path",
+                   type=str,
+                   required=False,
+                   default=argparse.SUPPRESS,
+                   help="Path to previously obtained calls to take as starting point")
+
+group.add_argument("--input_opt_path",
+                   type=str,
+                   required=False,
+                   default=argparse.SUPPRESS,
+                   help="(advanced) Path to exported optimizer state to take as the starting point")
+
+# add denoising config args
+# Note: we are hiding parameters that are either set by the model or are irrelevant to the case calling task
+gcnvkernel.DenoisingModelConfig.expose_args(
+    parser,
+    hide={
+        "--max_bias_factors",
+        "--psi_t_scale",
+        "--log_mean_bias_std",
+        "--init_ard_rel_unexplained_variance",
+        "--enable_bias_factors",
+        "--enable_explicit_gc_bias_modeling",
+        "--disable_bias_factors_in_active_class",
+        "--num_gc_bins",
+        "--gc_curve_sd",
+    })
+
+# add calling config args
+# Note: we are hiding parameters that are either set by the model or are irrelevant to the case calling task
+gcnvkernel.CopyNumberCallingConfig.expose_args(
+    parser,
+    hide={
+        '--p_active',
+        '--class_coherence_length',
+        '--initialize_to_active_class'
+    })
+
+# override some inference parameters
+gcnvkernel.HybridInferenceParameters.expose_args(parser)
+
+
+def update_args_dict_from_exported_model(input_model_path: str,
```

Comment:
A general comment: Use of type hinting is great but somewhat inconsistent.  Can you file an issue to go back and clean up?

Reply:
Will do :) it is hard to be too consistent since using most external modules (even `numpy`) breaks type inference. One conservative (though, verbose) strategy is to hint _everything_, as if we are coding in a statically typed language.